services/QuillLMS/app/controllers/api/v1/activity_sessions_controller.rb

```diff
@@ -103,12 +103,21 @@ def destroy
     data = params.delete(:data)&.permit!
     time_tracking = data && data['time_tracking']
     timespent = @activity_session&.timespent || ActivitySession.calculate_timespent(time_tracking)
+    max_4_bit_integer_size = 2147483647
+
+    if timespent && timespent > 3600
+      begin
+        raise "#{timespent} seconds for user #{@activity_session.user_id} and activity session #{@activity_session.id}"
+      rescue => e
+        Raven.capture_exception(e)
+      end
+    end
 
     params
       .permit(activity_session_permitted_params)
       .merge(data: data)
       .reject { |_, v| v.nil? }
-      .merge(timespent: timespent)
+      .merge(timespent: max_4_bit_integer_size > (timespent || 0) ? timespent : max_4_bit_integer_size)
```

Comment:
I think this is slightly more readable if we use `min`: `[(timespent || 0), max_4_bit_integer_size].min`

Reply:
I modified your version slightly to account for the case where timespent is actually nil, in which case I think we'd better save nil than 0. I don't think this should ever happen in reality because we have timetracking in all of our tools, but that's a closer preservation of the current behavior.